packages/protocol-kit/src/utils/passkeys/extractPasskeyData.ts

```diff
@@ -0,0 +1,68 @@
+import { Buffer } from 'buffer'
+import { PasskeyCoordinates, PasskeyArgType } from '../../types/passkeys'
+
+/**
+ * Extracts and returns the passkey data (coordinates and rawId) from a given passkey Credential.
+ *
+ * @param {Credential} passkeyCredential - The passkey credential generated via `navigator.credentials.create()` using correct parameters.
+ * @returns {Promise<PasskeyArgType>} A promise that resolves to an object containing the coordinates and the rawId derived from the passkey.
+ * @throws {Error} Throws an error if the coordinates could not be extracted
+ */
+export async function extractPasskeyData(passkeyCredential: Credential): Promise<PasskeyArgType> {
+  const passkey = passkeyCredential as PublicKeyCredential
+  const attestationResponse = passkey.response as AuthenticatorAttestationResponse
+
+  const publicKey = attestationResponse.getPublicKey()
+
+  if (!publicKey) {
+    throw new Error('Failed to generate passkey Coordinates. getPublicKey() failed')
+  }
+
+  const coordinates = await extractPasskeyCoordinates(publicKey)
+  const rawId = Buffer.from(passkey.rawId).toString('hex')
+
+  return {
+    rawId,
+    coordinates
+  }
+}
+
+/**
+ * Extracts and returns coordinates from a given passkey public key.
+ *
+ * @param {ArrayBuffer} publicKey - The public key of the passkey from which coordinates will be extracted.
+ * @returns {Promise<PasskeyCoordinates>} A promise that resolves to an object containing the coordinates derived from the public key of the passkey.
+ * @throws {Error} Throws an error if the coordinates could not be extracted via `crypto.subtle.exportKey()`
+ */
+export async function extractPasskeyCoordinates(
```

Comment:
It's at least used in this test
https://github.com/safe-global/safe-core-sdk/pull/859/files#diff-3d5ebe17b9fc5bb8f952e41a0496adc63ff2b8acad04c4061ac5ea59f92ec989R63

Reply:
yep! it's used in the `relay-key` tests and it allows users to extract the coordinates of the passkey when they have access to the public key. This can be useful because the passkey creation is handled by the DApp and we can provide different functions to create the `PasskeyArgType` object.